modules/core/src/main/java/org/apache/ignite/internal/cdc/CdcManager.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.ignite.plugin.PluginProvider;
+
+/**
+ * CDC manager responsible for logic of capturing data within Ignite node. Communication between {@link CdcManager} and
+ * {@link CdcMain} components is established through CDC management WAL records. {@link CdcManager} logs them, while {@link CdcMain}
+ * listens to them:
+ * <ul>
+ *     <li>
+ *         This manager can log {@link CdcManagerRecord} with committed consumer state. This record will be handled by
+ *         {@link CdcMain} the same way it handles {@link CdcConsumer#onEvents(Iterator)} when it returns {@code true}.
+ *     </li>
+ *     <li>
+ *         This manager must log {@link CdcManagerStopRecord} after it stopped handling records by any reason.
+ *         {@link CdcMain} will start consume records instead since {@link CdcManagerRecord#walState()} of
+ *         last consumed {@link CdcManagerRecord}.
+ *     </li>
+ *     <li>
+ *         Apache Ignite provides a default implementation - {@link CdcUtilityActiveCdcManager}. It is disabled from the
+ *         beginning, logs the {@link CdcManagerStopRecord} after Ignite node started, and then delegates consuming CDC
+ *         events to the {@link CdcMain} utility.
+ *     </li>
+ * </ul>
+ *
+ * Apache Ignite can be extended with custom CDC manager. It's required to implement this interface, create own {@link PluginProvider}
+ * that will return the implementation with {@link PluginProvider#createComponent(PluginContext, Class)} method.
+ * The {@code Class} parameter in the method is {@code CdcManager} class.
+ *
+ * @see CdcConsumer#onEvents(Iterator)
+ * @see CdcConsumerState#saveCdcMode(CdcMode)
+ */
+public interface CdcManager extends GridCacheSharedManager, DatabaseLifecycleListener {
+    /**
+     * If this manager isn't enabled then Ignite skips notifying the manager with following methods.
+     *
+     * @return {@code true} if manager is enabled, otherwise {@code false}.
+     */
+    public boolean enabled();
+
+    /**
+     * Callback is executed only once on Ignite node start when binary memory has fully restored and WAL logging is resumed.
+     * It's executed before the first call of {@link #collect(ByteBuffer)}.
+     * <p> Implementation suggestions:
+     * <ul>
+     *     <li>
+               It's required to register this manager to enable calling this method. The registration is performed with
+     *         {@link GridInternalSubscriptionProcessor#registerDatabaseListener(DatabaseLifecycleListener)}.
+     *     </li>
+     *     <li>Callback can be used for restoring CDC state on Ignite node start, collecting missed events from WAL segments.</li>
+     *     <li>Be aware, this method runs in the Ignite system thread and might lengthen the Ignite start procedure.</li>
+     *     <li>Ignite node will fail in case the method throws an exception.</li>
+     * </ul>
+     */
+    @Override public void afterBinaryMemoryRestore(IgniteCacheDatabaseSharedManager mgr,
```

Comment:
Can we keep default empty implementation of this methos? As it done in `DatabaseLifecycleListener`